tests/test_aio_function.py

```diff
@@ -44,6 +44,14 @@ async def test_decorator():
     assert await wrapper(1.0) == 1.0
     assert len(cache) == 2
 
+    # Calling the cache with None inserts it.
+    assert await wrapper(None) is None
```

Comment:
You don't check if the target was actually called or not. Same is true for the other cases.

Reply:
Good point - I added tests to check that the underlying function/method is only called when expected, in particular when None is the returned value.

I don't think it's necessary to add checks in all other cases however (I think it's safe to assume it will work for all non-None values, if it works for one non-None value, and None).